src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergFieldNode.java

```diff
@@ -0,0 +1,86 @@
+package com.snowflake.kafka.connector.internal.streaming.schemaevolution.iceberg;
+
+import java.util.LinkedHashMap;
+import java.util.stream.Collectors;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+class IcebergFieldNode {
+
+  public final String name;
+
+  public final String snowflakeIcebergType;
+
+  public final LinkedHashMap<String, IcebergFieldNode> children;
+
+  public IcebergFieldNode(String name, Type apacheIcebergSchema) {
+    this.name = name.toUpperCase();
```

Comment:
I am not sure about this `toUpperCase()`. At least fields inside the nested structures are case sensitive.
Even though we can wait for testing details like that once we set everything e2e.

Reply:
Interesting. Removing is not a problem. I'll keep it in mind.

Reply:
> At least fields inside the nested structures are case sensitive.

So then It won't work. I'll remove this `toUpperCase` method.